src/vm/eetoprofinterfaceimpl.cpp

```diff
@@ -3184,6 +3230,86 @@ HRESULT EEToProfInterfaceImpl::JITCompilationStarted(FunctionID functionId,
     }
 }
 
+HRESULT EEToProfInterfaceImpl::JITCompilationFinishedNoMetadata(FunctionID functionId)
+{
+    CONTRACTL
+    {
+        // Yay!
+        NOTHROW;
+
+        // Yay!
+        GC_TRIGGERS;
+
+        // Yay!
+        MODE_PREEMPTIVE;
+
+        // Yay!
+        CAN_TAKE_LOCK;
+
+        // The JIT / MethodDesc code likely hold locks while this callback is made
+
+        SO_NOT_MAINLINE;
+    }
+    CONTRACTL_END;
+    
+    CLR_TO_PROFILER_ENTRYPOINT((LF_CORPROF, 
+                                LL_INFO1000, 
+                                "**PROF: JITCompilationFinishedNoMetadata 0x%p.\n", 
+                                functionId));
+
+    _ASSERTE(functionId);
+
+    // Should only be called on profilers that support ICorProfilerCallback8
+    _ASSERTE(m_pCallback8 != NULL);
+
+    {
+        // All callbacks are really NOTHROW, but that's enforced partially by the profiler,
+        // whose try/catch blocks aren't visible to the contract system        
+        PERMANENT_CONTRACT_VIOLATION(ThrowsViolation, ReasonProfilerCallout);
+        return m_pCallback8->JITCompilationFinishedNoMetadata(functionId);
+    }
+}
+
+HRESULT EEToProfInterfaceImpl::JITCompilationStartedNoMetadata(FunctionID functionId)
+{
+    CONTRACTL
+    {
+        // Yay!
```

Comment:
Copy/pasted from this files source code, so they're pretty much there everywhere in this entire file.